### PR TITLE
add dtype for read functions

### PR DESCRIPTION
**Pull request by huidongchen**
This change is needed for [https://github.com/theislab/anndata/issues/75](url)

### CODE REVIEW

File: anndata/readwrite/read.py [hunk 1]
```diff
@@ -196,7 +198,7 @@ def read_mtx(filename: PathLike, dtype: str='float32') -> AnnData:
     X = mmread(fspath(filename)).astype(dtype)
     from scipy.sparse import csr_matrix
     X = csr_matrix(X)
-    return AnnData(X)
+    return AnnData(X,dtype=dtype)
```
Comment (falexwolf) [inline anndata/readwrite/read.py]: no space behind commas, please

File: anndata/readwrite/read.py [hunk 2]
```diff
@@ -55,14 +56,14 @@ def read_excel(
     """
     # rely on pandas for reading an excel file
     from pandas import read_excel
-    df = read_excel(fspath(filename), sheet)
-    X = df.values[:, 1:].astype(float)
+    df = read_excel(fspath(filename), sheet, dtype=dtype)
+    X = df.values[:, 1:].astype(dtype)
```
Comment (falexwolf) [inline anndata/readwrite/read.py]: why do we need the dtype in both lines? it should not be necessary in the second line anymore, should it?

**Comment by falexwolf** [comment]
Do the tests run through on your end? Can you add a test, please?

I believe that the fact that one test currently fails is not due to your PR; are you sure you rebased properly? It went through on master.

**Comment by falexwolf** [comment]
Oh damn, the test failing due to the "open file" is my fault. It went through for me but I'll look into this, ignore this comment.

**Comment by falexwolf** [comment]
OK, I just fixed the test on master... nothing to do with anything you'd do.

Thank you for your PR! :smile:

File: anndata/readwrite/read.py (hunk 2 above)
Comment (huidongchen) [reply anndata/readwrite/read.py]: Agreed. Will remove it.

**Comment by huidongchen** [comment]
hmm, it passed all the local testing on my end. Have no idea why it still failed..

**Comment by falexwolf** [comment]
Thanks, it's fine!

**Pull request merged**